pkg/api/v1alpha1/cluster_types.go

```diff
@@ -1211,6 +1248,12 @@ func (c *Cluster) MachineConfigRefs() []Ref {
 	return machineConfigRefMap.toSlice()
 }
 
+// SetFailure sets the failureMessage and failureReason of the Cluster status.
+func (c *Cluster) SetFailure(failureMessage *string, failureReason *string) {
```

Comment:
this is a helper, so why wouldn't we just take strings and do the pointer conversion inside?
it guarantees not nil pointers and also it makes using this helper easier

Reply:
Updated to take a string and a `FailureReasonType`. Also added a `ClearFailure` helper for clearing failures.